121 design how to get form creators to provide a url for a page with privacy information for their form

deberny opened the pull request:
[Trello card](https://trello.com/c/5hf54nQf)

[Design based on Mural area](https://app.mural.co/t/gaap0347/m/gaap0347/1652697480711/fa60d6e36c472088d3c49ea4a7e7fc5b1d8186d4?wid=0-1659598224882)

# Relevant user story

As a form creator, I need to create a privacy notice for my form, so the people completing it know why we're asking for the information how we will process it, and to meet GDPR.

## Why are we working on this user story this sprint?

- The department that uses GOV.UK Forms is the ‘data controller’ and they have to provide information about the data they collect and how they manage it to meet the requirements of GDPR.

## Hypothesis

**If we** ask the form creator to provide a link to privacy information for a form before they can make a form live
**then** they will be able to provide a ink to privacy information for their form
**then** including privacy notice will become part of the form creation process and we can provide a link to the privacy information alongside the form

**If we** include guidance about what privacy information needs to be provided and why it is needed
**then** users will feel more confident in what it needs to cover, how to provide a link to one or how to get one created.

## Test

We’ll proceed if users are confident in providing a link to privacy information for their form

# Page changes

## New page to add privacy information
![screencapture-localhost-3000-form-designer-provide-link-to-privacy-information-2022-08-11-16_47_21](https://user-images.githubusercontent.com/2632224/184176286-5cae5939-5383-457d-b171-029c6d44cd7d.png)

## Fixes to Make your form live page
To address previous PR 😄 

![screencapture-localhost-3000-form-designer-make-your-form-live-2022-08-11-16_48_40](https://user-images.githubusercontent.com/2632224/184176878-b8c118e3-1acb-4c3a-beca-7e516459db80.png)

hannahkc commented:
Both pages look good to me. My only question is if we think we need hint text to help people with the format of the URL? Or if we add a http:// prefix in the URL field - like the suggestion on the Mural? Though I guess it could be a URL that starts with https:// and we also probably want to be careful about making people think they can't just copy and paste the whole address into the field? hmmmm... 

Maybe we could add hint text that says: For example, https://www.gov.uk/help/privacy-notice

Gives an example if a privacy notice too which might be useful. What do you think?

christophercameron-ixd reviewed (approved):
Looking good to me!

deberny commented:
Good point @hannahkc I'll add the hint text. I also forgot to include the Privacy link in the 'Form preview' footer

deberny commented:
Updated with the hint text example. Defo better @hannahkc 

![screencapture-localhost-3000-form-designer-provide-link-to-privacy-information-2022-08-12-16_52_26](https://user-images.githubusercontent.com/2632224/184395323-c70a621d-d52b-44c6-aa47-4c37518cff4d.png)

Added footer links to form preview page. Couldn't figure out how to pass user input data into the href so it goes nowhere atm. We can add it later 🥲 

<img width="1178" alt="Screenshot 2022-08-12 at 16 52 00" src="https://user-images.githubusercontent.com/2632224/184395044-38ee987e-27d4-4f20-8250-b2a1a8d0b537.png">

I've tried to display the form name in the form preview as well, but no success. Will ask dev team for advice 😉
